### ISSUE TITLE
info on CamemBERT preprocessing

### ISSUE THREAD

**Issue by LeoDeep** [pytorch/fairseq#1501]
Hello,
we are trying to train a new RoBERTa model from scratch following the CamemBERT paper [https://arxiv.org/abs/1911.03894](url).

Currently we have done the following steps:

1. Downloading our dataset for training RoBERTa
2. Creating our sentencepiece.bpe.model and sentencepiece.bpe.vocab following `RAW_DATA_FPATH = "dataset.txt"
MODEL_PREFIX = "tokenizer" 
VOC_SIZE = 32000
SUBSAMPLE_SIZE = 12800000 
NUM_PLACEHOLDERS =  0
SPM_COMMAND = ("--input={} --model_prefix={} "
               "--vocab_size={} --input_sentence_size={} "
               "--shuffle_input_sentence=true "
               "--model_type=bpe").format(
               RAW_DATA_FPATH, MODEL_PREFIX, 
               VOC_SIZE - NUM_PLACEHOLDERS, SUBSAMPLE_SIZE)
spm.SentencePieceTrainer.Train(SPM_COMMAND)`
3. We split the dataset in dataset_train.txt, dataset_val.txt, dataset_test.txt
4. We encode the previous txt files in the bpe format in order to pass them to roBERTa based on the following snippet:
`for SPLIT in train val test;
do 
cat dataset_${SPLIT}.txt | spm_encode --model=sentencepiece.bpe.model --output_format=piece > dataset_${SPLIT}.bpe
done`

In order to run correctly the **fairseq-preprocessing** and then **fairseq-train** (following [https://github.com/pytorch/fairseq/blob/master/examples/roberta/README.pretraining.md](url) ) do we need to add the _bos_ and _eos_ token with the **spm_encode** parameter **--extra_options=bos:eos** ?

Thanks in advance.

**Comment by lematt1991** [comment]
[Fairseq will add these tokens automatically](https://github.com/pytorch/fairseq/blob/master/fairseq/data/dictionary.py#L23-L25)
